### ISSUE TITLE
$all on array of nested documents does not work

### ISSUE THREAD

**Issue by j** [crcn/sift.js#160]
Using query: `{ 'order.items.product': { $all: ['poster', 'frame'] } }`

```javascript

const items = [{
	"order" : {
		"id" : "or_0001",
		"amount" : 6000,
		"items" : [
			{
				"product" : "poster",
				"sku" : "P18x24",
				"quantity" : 1,
				"amount" : 3000
			},
			{
				"product" : "frame",
				"sku" : "P18x24",
				"quantity" : 1,
				"amount" : 3000
			},
			{
				"product" : "shipping",
				"sku" : "shipping",
				"quantity" : 1,
				"amount" : 5000
			}
		]
	}
}];

items.filter(sift({ 'order.items.product': { $all: ['poster', 'frame'] } });
```

This query works using mongo, but not sift.

**Comment by NicoBeyer** [comment]
I second that. But it has nothing to d with $all. It doesn't properly filter complex array values.

This also fails:
`const obj = {
            array: [
                {value: 1},
                {value: 2},
            ],
        };
        const query  = {
            "array.value": 1
        };
        const filter = sift(query);

        const res = filter(obj);

        assert.isTrue(res);`

**Comment by crcn** [comment]
Sorry for not getting to this sooner, this bug required a whole bunch of refactoring. Issue should be fixed in `11.0.0` 👍.
